Mathlib/Tactic/NormNum/Pow.lean

```diff
@@ -66,27 +66,26 @@ a proof of depth `log (log b)` which will essentially never overflow before the
 themselves exceed memory limits.
 -/
 partial def evalNatPow (a b : Q(ℕ)) : (c : Q(ℕ)) × Q(Nat.pow $a $b = $c) :=
-  if b.natLit! = 0 then
+  if b.natLit! = 0 then -- a^0 = 1
     haveI : $b =Q 0 := ⟨⟩
     ⟨q(nat_lit 1), q(natPow_zero)⟩
-  else if a.natLit! = 0 then
+  else if a.natLit! = 0 then -- 0^b = 0
     haveI : $a =Q 0 := ⟨⟩
     have b' : Q(ℕ) := mkRawNatLit (b.natLit! - 1)
     haveI : $b =Q Nat.succ $b' := ⟨⟩
     ⟨q(nat_lit 0), q(zero_natPow)⟩
-  else if a.natLit! = 1 then
+  else if a.natLit! = 1 then -- 1^b = 1
     haveI : $a =Q 1 := ⟨⟩
     ⟨q(nat_lit 1), q(one_natPow)⟩
-  else if b.natLit! = 1 then
+  else if b.natLit! = 1 then -- a^1 = a
     haveI : $b =Q 1 := ⟨⟩
     ⟨a, q(natPow_one)⟩
   else
-    let ⟨c, p⟩ := go b.natLit!.log2 a (mkRawNatLit 1) a b _ .rfl
+    let ⟨c, p⟩ := go b.natLit!.log2 a (mkRawNatLit 1) a b
     ⟨c, q(($p).run)⟩
 where
-  /-- Invariants: `a ^ b₀ = c₀`, `depth > 0`, `b >>> depth = b₀`, `p := Nat.pow $a $b₀ = $c₀` -/
```

Comment:
It's possible that these changes are bad and that `PowMod.lean` should also be changed, see [this comment](https://github.com/leanprover-community/mathlib4/pull/9015#issuecomment-2136264087). Benchmarking may be required.

Reply:
I reverted to existing pattern since that I don't have to think about it. We can try some benchmarking the lack of indirection in a later PR.